sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/TargetDialog.java

```diff
@@ -10,192 +9,189 @@
+    }
+
+    public CallIdentifier getCallIdentifer() {
+        return callIdentifier;
+    }
+
+    /**
+     * set the CallId field
+     * @param cid String to set. This is the body part of the Call-Id
+     *  header. It must have the form localId@host or localId.
+     * @throws IllegalArgumentException if cid is null, not a token, or is
+     * not a token@token.
+     */
+    public void setCallId(String cid) {
+        callId = cid;
+    }
+
+    /**
+     * Set the callIdentifier member.
+     * @param cid CallIdentifier to set (localId@host).
+     */
+    public void setCallIdentifier(CallIdentifier cid) {
+        callIdentifier = cid;
+    }
+
+    /**
+     * Gets the local-tag parameter from the address param list.
+     * @return tag field
+     */
+    public String getLocalTag() {
+        if (parameters == null)
+            return null;
+        return getParameter(ParameterNames.LOCAL_TAG);
+    }
+    /**
+     * Sets the local tag of the Target-Dialog header.
+     *
+     * @param localTag the local tag value to set
+     * @throws ParseException if the provided localTag is invalid
+     */
+    public void setLocalTag(String t) throws ParseException {
+    	  if (t == null)
+              throw new NullPointerException("null tag ");
+          else if (t.trim().equals(""))
+              throw new ParseException("bad tag", 0);
+          this.setParameter(ParameterNames.LOCAL_TAG, t);
+    }
+    /** Boolean function
+     * @return true if the Tag exist
+     */
+    public boolean hasLocalTag() {
+        return hasParameter(ParameterNames.LOCAL_TAG);
+    }
+
+    /** remove Tag member
+     */
+    public void removeLocalTag() {
+        parameters.delete(ParameterNames.LOCAL_TAG);
+        
+    }
+    /**
+     * Gets the remote tag of the Target-Dialog header.
+     *
+     * @return the remote tag value
+     */
+    public String getRemoteTag() {
+    	 if (parameters == null)
+             return null;
+         return getParameter(ParameterNames.REMOTE_TAG);
+    }
+    /**
+     * Sets the remote tag of the Target-Dialog header.
+     *
+     * @param remoteTag the remote tag value to set
+     * @throws ParseException if the provided remoteTag is invalid
+     */
+    public void setRemoteTag(String t) throws ParseException {
+    	  if (t == null)
+              throw new NullPointerException("null tag ");
+          else if (t.trim().equals(""))
+              throw new ParseException("bad tag", 0);
+          this.setParameter(ParameterNames.REMOTE_TAG, t);
+    }
+    /** Boolean function
+     * @return true if the Tag exist
+     */
+    public boolean hasRemoteTag() {
+        return hasParameter(ParameterNames.REMOTE_TAG);
+    }
+
+    /** remove Tag member
+     */
+    public void removeRemoteTag() {
+        parameters.delete(ParameterNames.REMOTE_TAG);      
+    }
+
+    @Override
+    public void setValue(String value) throws ParseException {
+        throw new ParseException(value,0);
```

Comment:
I think this one can call decodeBody instead of throwing an exception, no ?

Reply:
You're absolutely right! Instead of throwing an exception for missing local or remote tags, the decodeBody method can handle them gracefully by checking for their existence before attempting to parse the values. Fixed it